src/lib_json/json_value.cpp

```diff
@@ -104,7 +111,11 @@ static inline char* duplicateStringValue(const char* value,
  */
 static inline char* duplicateAndPrefixStringValue(
```

Comment:
Note that this is file-static. That means we can change the signature of this without worrying about binary-compatibility. We don't necessarily need ifdefs for this.

But why do we need to change the signature? I think you've misunderstood the purpose. Because we needed to store string-lengths but we could not add a field, we have encoded the length onto the front of these "allocated" strings. It's a hack. But it should already provide all the information you need. When you release, just call decodePrefix first to get the length.